explorer/src/lib/services/duskAPI.js

```diff
@@ -230,12 +230,26 @@ const duskAPI = {
 
   /**
    * @param {string} id
-   * @returns {Promise<Transaction>}
+   * @returns {Promise<Transaction | String | Error>}
```

Comment:
The promise never resolves with an `Error`

Reply:
I'm not really keen on having the promise resolve with a string here: I would have thrown a custom error instead.

That being said: it'll ease our job (less refactoring) if we plan to show partial data as a near future improvement.
So... leave as it is if that's the plan.

Reply:
Also, the transaction being in the mempool isn't an error, it's part of the protocol.

Reply:
Yes, of course, but here we're retrieving a spent transaction, in theory, and we haven't found one.
So if the code were as it is I would return a custom error.

If we plan to return a "partial transaction" in future then it makes sense to resolve the promise with such data.

What I meant is that if we plan to add the partial data thing, then I can live with resolving with a string now, as we'll have less code to touch later.